rust/lance/src/index.rs

```diff
@@ -346,57 +348,65 @@ impl DatasetIndexExt for Dataset {
         Ok(())
     }
 
-    async fn index_statistics(&self, index_name: &str) -> Result<Option<String>> {
-        let index_uuid = self
-            .load_index_by_name(index_name)
-            .await?
-            .map(|idx| idx.uuid.to_string());
-
-        if let Some(index_uuid) = index_uuid {
-            let index_statistics = self
-                .open_generic_index("vector", &index_uuid)
-                .await?
-                .statistics()?;
-            Ok(Some(index_statistics))
-        } else {
-            Ok(None)
+    async fn index_statistics(&self, index_name: &str) -> Result<serde_json::Value> {
+        let metadatas = self.load_indices_by_name(index_name).await?;
+        if metadatas.is_empty() {
+            return Err(Error::IndexNotFound {
+                identity: format!("name={}", index_name),
+                location: location!(),
+            });
         }
-    }
-
-    async fn count_unindexed_rows(&self, index_uuid: &str) -> Result<Option<usize>> {
-        let index = self.load_index(index_uuid).await?;
 
-        if let Some(index) = index {
-            let unindexed_frags = unindexed_fragments(&index, self).await?;
-            let unindexed_rows = unindexed_frags
-                .iter()
-                .map(Fragment::num_rows)
-                // sum the number of rows in each fragment if no fragment returned None from row_count
-                .try_fold(0, |a, b| b.map(|b| a + b).ok_or(()));
+        let column = self
+            .schema()
+            .field_by_id(metadatas[0].fields[0])
+            .map(|f| f.name.as_str())
+            .ok_or(Error::IndexNotFound {
+                identity: index_name.to_string(),
+                location: location!(),
+            })?;
 
-            Ok(unindexed_rows.ok())
-        } else {
-            Ok(None)
-        }
-    }
+        // Open all delta indices
+        let deltas = stream::iter(metadatas.iter())
+            .then(|m| async move { self.open_generic_index(column, &m.uuid.to_string()).await })
+            .try_collect::<Vec<_>>()
+            .await?;
 
-    async fn count_indexed_rows(&self, index_uuid: &str) -> Result<Option<usize>> {
-        let count_rows = self.count_rows();
-        let count_unindexed_rows = self.count_unindexed_rows(index_uuid);
+        // Stastistics for each delta index.
+        let delta_stats = deltas
+            .iter()
+            .map(|idx| idx.statistics())
+            .collect::<Result<Vec<_>>>()?;
 
-        let (count_rows, count_unindexed_rows) =
-            futures::try_join!(count_rows, count_unindexed_rows)?;
+        let unindexed_fragments = self.unindexed_fragments(index_name).await?;
+        let num_unindexed_rows = unindexed_fragments
+            .iter()
+            .map(|f| f.num_rows().unwrap_or_default())
```

Comment:
`unwrap_or_default` will return `0` for any `None`.  We should probably error out instead if it is `None`.